pkg/pod/pod.go

```diff
@@ -0,0 +1,84 @@
+package pod
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodPredicate func(corev1.Pod) bool
+
+func FilterPods(podList *corev1.PodList, predicates ...PodPredicate) *corev1.PodList {
+	filteredPods := &corev1.PodList{}
+	for _, pod := range podList.Items {
+		var match = true
+		for _, p := range predicates {
+			if !p(pod) {
+				match = false
+				break
+			}
+		}
+		if match {
+			filteredPods.Items = append(filteredPods.Items, pod)
+		}
+	}
+
+	return filteredPods
+}
+
+type DeleteResult struct {
+	Message              string
+	NumMarkedForDeletion int
+}
+
+func DeletePods(c client.Client, pl *corev1.PodList) (*DeleteResult, error) {
+	me := &multierror.Error{}
+	var podsMarkedForDeletion []string
+	for _, p := range pl.Items {
+		if p.DeletionTimestamp == nil {
+			err := c.Delete(context.TODO(), &p)
```

Comment:
Thoughts on setting a `TerminationGracePeriodSeconds` to 0 in the `Spec` before issuing the delete?
Sometimes that is the only last resort to get rid of a stubborn pod.